src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculator.php

```diff
@@ -55,4 +66,23 @@ public function calculateOriginal(ProductVariantInterface $productVariant, array
 
         throw MissingChannelConfigurationException::createForProductVariantChannelPricing($productVariant, $context['channel']);
     }
+
+    public function calculateLowestPriceBeforeDiscount(ProductVariantInterface $productVariant, array $context): ?int
+    {
+        Assert::keyExists($context, 'channel');
+        $channel = $context['channel'];
+        Assert::isInstanceOf($channel, ChannelInterface::class);
+
+        /** @var ChannelPricingInterface|null $channelPricing */
+        $channelPricing = $productVariant->getChannelPricingForChannel($channel);
+        if (null === $channelPricing) {
+            throw MissingChannelConfigurationException::createForProductVariantChannelPricing($productVariant, $channel);
+        }
+
+        if (!$this->productVariantLowestPriceDisplayChecker->isLowestPriceDisplayable($productVariant, $context)) {
```

Comment:
Since `productVariantLowestPriceDisplayChecker` can be null we should check for that

Reply:
Good catch